.github/mergify.yml

```
@@ -4,14 +4,14 @@ defaults:
       method: squash
       commit_message_template: |
         {{ title }}
-        
+
         {{ body | get_section("## Description", "") }}
-        
+
         Pull-Request: #{{ number }}.
         {# Here comes some fancy Jinja2 stuff for correctly attributing co-authorship: #}
         {% for commit in (commits | unique(False, 'email_author')) | rejectattr("author", "==", author) %}
           {% if commit.parents|length == 1 %}
-        Co-Authored-By: {{ commit.author }} <{{ commit.email_author }}>
+        Co-authored-by: {{ commit.author }} <{{ commit.email_author }}>
           {% endif %}
         {% endfor %}
```

The following (note the `-`) should remove all the newlines between the `Co-authored-by` lines:

```suggestion
        {%- for commit in (commits | unique(False, 'email_author')) | rejectattr("author", "==", author) -%}
          {%- if commit.parents|length == 1 -%}
        Co-authored-by: {{ commit.author }} <{{ commit.email_author }}>
          {% endif %}
        {%- endfor -%}
```

---

Tested using https://j2live.ttl255.com/ and the following test-data:

```yml
commits:
  - email_author: foo@example.com
    author: foo
    parents:
      - abc
  - email_author: bar@example.com
    author: bar
    parents:
      - abc
  - email_author: bar@example.com
    author: bar
    parents:
      - abc
  - email_author: xyz@example.com
    author: xyz
    parents:
      - abc
      - def
  - email_author: foo@example.com
    author: foo
    parents:
      - abc
  - email_author: xyz@example.com
    author: xyz
    parents:
      - abc
  - email_author: xyz1@example.com
    author: xyz1
    parents:
      - abc
author: test
```

---

I also noticed that the code still has a bug. If the first commit of a contributor is a merge-commit, we won't attribute any further commits of them because the `unique` already filtered them. We need to filter out merge commits first, I am working on a fix.